cmd/share-secret/main.go

```diff
@@ -37,19 +41,20 @@ func run() error {
 
 func PostSecret() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		input := struct {
-			Secret string
-			Unit   string
-			Time   int64
-		}{}
+		var input StoreData
+
 		if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
+
+		keyData := input.GetKey()
+		mData[keyData] = input
+
 		data, err := json.Marshal(&struct {
 			URI string `json:"uri"`
 		}{
-			URI: "https://example.com/secret/123456789",
+			URI: "https://localhost:8080/secret/" + keyData,
```

Comment:
Because we don't have a ssl certificate `https` is not working. It should be `http` instead